### ISSUE TITLE
Boleto Validator - Support "convenio" and return which partial is wrong

### ISSUE THREAD

**Issue by renatoagds** [brazilian-utils/brazilian-utils#55]
This issue propose 2 features:

1) Add support for "convenio" in "boleto" validator, since it related formatter ([`format-boleto`](https://github.com/brazilian-utils/brazilian-utils/tree/master/packages/format-boleto)) supports "convenio"

2) Returns which partial is wrong in [`is-valid-boleto`](https://github.com/brazilian-utils/brazilian-utils/tree/master/packages/is-valid-boleto). To avoid break the current contract, we could just add a second parameter, that will be a flag to turn on this feature.

**Comment by renatoagds** [comment]
cc/ @hyanmandian

**Comment by hyanmandian** [comment]
Hello @renatoagds, we are working in a fork project to refactoring the entire project and we plan to create 2 separated functions for that: `format-bank-slip` and `format-covenant` (I really don't know if these are the best names for that functions). If you need to format both cases, you can just compose these functions (or maybe we can create another function for that).

About `is-valid-boleto`, I think that we can fix this issue together with the refactoring.

Today I will create a branch with this fork and you could work with us :D

Thanks for the ideas and sorry for the delay.

**Comment by hyanmandian** [comment]
Hey @renatoagds can you check *next* branch and confirm if we fixed your issue?

**Comment by renatoagds** [comment]
@hyanmandian for sure! will test today and let you know.

**Comment by renatoagds** [comment]
@hyanmandian Confirming that's not the expected. 

1) The `next` branch doesn't include support for `convenio`, in `format` and `isValid` (this is actually a regression, since `format` in `master` supports `convenio`).

2) The current `isValid` for `boleto` doesn't return which block is wrong, it only returns false/true. 

Can I help with the requested features in next?

**Comment by hyanmandian** [comment]
Yes, I can help! What you think to add a second param to "config" the expected return? Something like:

```
isValidBoleto("1541541587487") // false
isValidBoleto("1541541587487", { block: true }) // [false, false, false, false, false, false]
```

I think that is better because we don't break our current API and it's work for all packages we have.

**Comment by hyanmandian** [comment]
About `convenio` I think we need to create another package to do that and another package to format and validate `boleto` and `convenio` together, something like `isValidBoletoCovenant` or `isValidBoletoConvenio` and don't know a good name for `convenio` :(

**Comment by renatoagds** [comment]
> What you think to add a second param to "config" the expected return?

That's exactly what I was thinking about it, since I put the same idea in issue description. We could call this new param `options` maybe.

> About convenio I think we need to create another package to do that

Why do you think this is necessary? That's some case that wouldn't use `convenio`? I think we could split the internal logic between `convenio` and conventional `boleto`, but in the final package, it should be exposed just one function, that detects if starts with `8` and switch between handlers.

**Comment by hyanmandian** [comment]
I think is better to have 3 packages: `boleto`, `convenio` and `boletoOrConvenio` (we need to think in better names, ok? HahahHAh). In that way, we cover all use cases, what you think?

**Comment by renatoagds** [comment]
> In that way, we cover all use cases, what you think?

Sounds good! 🚀
